kong/init.lua

```diff
@@ -881,23 +881,20 @@ function Kong.init_worker()
   end
 
   if kong.clustering then
-    if is_control_plane(kong.configuration) then-- CP needs to support both full and incremental sync
-      kong.clustering:init_worker()
-    
-    -- full sync is only enabled for DP if incremental sync is disabled
-    elseif is_data_plane(kong.configuration) and not kong.sync then
-      local using_dedicated = kong.configuration.dedicated_config_processing
-      if using_dedicated and process.type() == "privileged agent" then
-        -- full sync dp agent
-        kong.clustering:init_worker()
-        return
+    local is_cp = is_control_plane(kong.configuration)
+    local is_dp_sync_v1 = is_data_plane(kong.configuration) and not kong.sync
+    local using_dedicated = kong.configuration.dedicated_config_processing
 
-      end
+    -- CP needs to support both full and incremental sync
+    -- full sync is only enabled for DP if incremental sync is disabled
+    if is_cp or is_dp_sync_v1 then
```

Comment:
In such case, with dedicated_config_processing enabled, the normal workers will also start `clustering:init_worker()`, which is unnecessary.

Reply:
See https://github.com/Kong/kong/blob/master/kong/clustering/init.lua#L85-L92, DP will only run `init_worker()` in `worker 0` or `privileged agent`

Reply:
Referencing [this section in init.lua](https://github.com/Kong/kong/blob/master/kong/clustering/init.lua#L85-L92) and [this section in utils.lua](https://github.com/Kong/kong/blob/master/kong/clustering/utils.lua#L162-L169), DP will only start the clustering process in Worker 0 and the privileged process.

Reply:
@StarlightIbuki , do you have some idea about this?